### PR TITLE
Minting contract

### LINKED ISSUE #8: Implement metadata for purchase transaction.
When purchasing an NFT, the metadata on the transaction must include the various information needed to recreate the token name and currency symbol.
This information is defined [here](https://drive.google.com/file/d/1F7a0lOWKJhEuYZxcK90qN2UN1rVePXRX/view).
You must modify `Seabug.Contract.MarketPlaceBuy` to include the `727` standard defined in the above document. DO NOT implement the delta standard, as this is a future change.

### LINKED ISSUE #21: Implement contract for minting an NFT

### PR DESCRIPTION
**Pull request by rynoV**
Closes #21. Also closes #8 and purchase metadata is tested and working with this branch. All changes from #10 are included here because they were used for minting. Merging into calum/purchase-metadata to make it clearer what changes were for metadata and what was for minting.

This PR also includes changes to the parsing of the metadata received from blockfrost, because the nfts minted via CTL have a different metadata format. So the previously minted nfts still on the marketplace won't work after this PR.

### CODE REVIEW

File: README.md
Comment (samuelWilliams99) [inline README.md:15]: we're using branch specific links here, we'll want to change this to main when we merge up

File: index.js
Comment (samuelWilliams99) [inline index.js:39]: lets put a comment with this so it doesn't get removed

File: src/Seabug/CallContract.purs
Comment (samuelWilliams99) [inline src/Seabug/CallContract.purs:87]: maybe specify sgNft here instead

File: src/Seabug/Contract/MarketPlaceListNft.purs
Comment (t4ccer) [reply src/Seabug/Contract/MarketPlaceListNft.purs:71]: When price is lower than min ADA you can't really pay

File: src/Seabug/Types.purs
Comment (t4ccer) [reply src/Seabug/Types.purs:68]: Yes, to derive hash of fee vault script

File: src/Seabug/CnftMintPolicy.purs
Comment (samuelWilliams99) [inline src/Seabug/CnftMintPolicy.purs:42]: is this needed for deployment? Looks like it just calculates the currency symbol?
if we do need this `main`, can we pass in the txoutref as commandline args?

Comment (rynoV) [reply src/Seabug/CnftMintPolicy.purs]: Forgot to remove this

File: README.md
Comment (rynoV) [reply README.md:15]: These are permalinks so they should continue to work (I think). The lines will probably get moved around, but this seems like a good way to do this since it should be easy to figure out what lines need to be acted on by comparing the old linked version with the updated version

File: src/Seabug/Contract/MarketPlaceListNft.purs
Comment (rynoV) [reply src/Seabug/Contract/MarketPlaceListNft.purs:71]: I'll just update this to use our `minAdaOnlyUTxOValue` then

File: flake.nix
Comment (samuelWilliams99) [inline flake.nix:46]: Is this correct? does this mean we're no longer bundling for the build?

File: src/Seabug/Contract/MarketPlaceListNft.purs
Comment (samuelWilliams99) [inline src/Seabug/Contract/MarketPlaceListNft.purs:71]: Can we verify a 2ada nft can still be bought?

File: src/Seabug/Contract/Mint.purs
Comment (samuelWilliams99) [inline src/Seabug/Contract/Mint.purs:62]: Verify this won't be a breaking change, I believe the onchain is coded such that staking addresses aren't permitted right now. We'll need to update that to merge this if this breaks, or move this out to another PR

File: src/Seabug/Contract/MarketPlaceListNft.purs
Comment (rynoV) [reply src/Seabug/Contract/MarketPlaceListNft.purs:71]: Confirmed that a `minAdaOnlyUTxOValue` (actually 1 ada) priced nft can be bought, this is the purchase transaction: https://testnet.cardanoscan.io/transaction/dace69f58dc611579bd5c7a7cf0de1d4b307f11f9fce01663bc7799cc6fd799c

File: src/Seabug/Contract/Mint.purs
Comment (rynoV) [reply src/Seabug/Contract/Mint.purs:62]: Minting the token purchased in the above transaction was done with this change, so it seems okay?

File: flake.nix
Comment (samuelWilliams99) [reply flake.nix:46]: Ah okay, I see.

File: src/Seabug/Contract/Mint.purs
Comment (samuelWilliams99) [reply src/Seabug/Contract/Mint.purs:62]: Huh, the source code really looked like it wouldn't allow this, well good to know

File: src/Seabug/Contract/MarketPlaceListNft.purs
Comment (samuelWilliams99) [reply src/Seabug/Contract/MarketPlaceListNft.purs:71]: neato!

**Review by samuelWilliams99 — APPROVED** [review]

File: src/Seabug/Contract/Mint.purs
Comment (t4ccer) [reply src/Seabug/Contract/Mint.purs:62]: Staking (enterprise) addresses aren't allowed when changing owner iirc, should work fine for minting. Payment for changing owner has to be sent to address without staking key though.

**Comment by rynoV** [comment]
Verified fetching a single nft still works after merge

File: src/Seabug/Contract/Mint.purs
Comment (samuelWilliams99) [reply src/Seabug/Contract/Mint.purs:62]: How did the 1 ada cost buy transaction pass then? Was the ownerstake perhaps still Nothing?

Comment (rynoV) [reply src/Seabug/Contract/Mint.purs:62]: @samuelWilliams99 I think because the buy contract doesn't use a staking key, it uses the `ownPaymentPubKeyHash` of the buyer

Comment (samuelWilliams99) [reply src/Seabug/Contract/Mint.purs:62]: oh of course! Sorry thats my bad - I thought this change was in the buy contract, but i see its for minting 😅 
Okay, Im happy with this then, lets merge :D

**Pull request merged**
